src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -78,13 +98,47 @@ constructor a_fresh statements
    = Nothing
 
   pair x
+   | XValue _ (PairT ta tb) (VPair a b) <- x
+   = Just (ta, tb, xValue ta a, xValue tb b)
+
    | Just (PrimMinimal (Min.PrimConst (Min.PrimConstPair ta tb)), [a,b]) <- takePrimApps x
-   = Just (ta,tb,a,b)
+   = Just (ta, tb, a, b)
+
    | otherwise
    = Nothing
 
+  option x
+   | XValue _ (OptionT tv) VNone <- x
+   = Just (tv, xFalse, xDefault tv)
+
+   | Just (PrimOption (PrimOptionPack tv), [b, v]) <- takePrimApps x
+   = Just (tv, b, v)
+
+   | Just (PrimMinimal (Min.PrimConst (Min.PrimConstSome tv)), [v]) <- takePrimApps x
+   = Just (tv, xTrue, v)
+
+   | otherwise
+   = Nothing
+
+
   get env n
    | (_,x'):_ <- filter ((==n).fst) env
    = Just x'
    | otherwise
    = Nothing
+
+  -- ANF? :(
```

Comment:
I thought arguments to prim apps would definitely be xvars.... perhaps I misunderstood?